src/manifest.c

```diff
@@ -548,6 +536,16 @@ static int retrieve_manifests(int current, int version, char *component, struct
 		goto out;
 	}
 
+	/* Only signing the MoM manifest*/
+	if (strncmp(component, "MoM", 3) == 0) {
+		free(filename);
+		string_or_die(&filename, "%s/%i/Manifest.MoM", state_dir, version);
+		if (verify_signatures && !download_and_verify_signature(url, filename)) {
```

Comment:
Can you check to see if Manifest.MoM.sig exists, and use the local copy if possible? If it's not present, then we can `download_and_verify_signature()`.